### PR TITLE
Use curl instead of HurlCli

### PR DESCRIPTION
**Pull request by robbytx**
This pull request rewrites the ``hurl`` script to actually invoke ``curl``, rather than the often-incompatible Java class.

### CODE REVIEW

File: cli/hurl [hunk 1]
```diff
@@ -21,59 +24,128 @@ OPTIONS:
...
+        ;;
+        -J)
+            append_args -H 'Content-Type: application/json'
+        ;;
+        -d|--data|--data-ascii)
+            DEFAULT_METHOD=POST
+            DATA="$1"
+            shift
+        ;;
+        -d|--data|--data-binary)
```
Comment (robbytx) [inline cli/hurl]: ``-d`` and ``--data`` are redundant and won't match.

File: cli/hurl [hunk 2]
```diff
@@ -21,59 +24,128 @@ OPTIONS:
...
+        -G|--get)
+            METHOD=GET
+            append_args "$key"
+        ;;
+        -I|--head)
+            METHOD=HEAD
+            append_args "$key"
+        ;;
+        -X*) # -XPOST
+            METHOD=$(echo $key |cut -f3-)
```
Comment (robbytx) [inline cli/hurl]: This should be ``cut -c3-``

**Comment by jsurls** [comment]
@robbytx is the goal to create a hurl client built on curl not using the java code at all?

If so should we make a separate repo for this? I still see value in having a cli tool to execute the built jar for convenience testing but for everyday usage I'd prefer to have a separate command that doesn't necessitate the need to build that jar.

**Comment by robbytx** [comment]
@jsurls My main goal was to provide a drop-in replacement for curl that supports access to servers that require hmac-auth.  ``HurlCli`` is only compatible with curl in the narrowest sense -- for specific command-line arguments, and only when the server returns a 2xx response.  These incompatibilities (especially the lack of visibility into non-2xx responses) make it frustrating to use, when you are used to the power and transparency that curl offers.

I'm perfectly happy to delete the Java-based implementation entirely, but if you see that there is value in retaining it (for the purposes of one-off testing of the client library), then I see two options:
 1. We can move HurlCli into ``client/src/test/java/`` so it can be invoked there.
 2. We can move this script up to the top-level of the project (or a /bin/ subdirectory) and update the READMEs accordingly.

To be honest, I didn't do either of these yet, because I wasn't sure whether we wanted to keep HurlCli around or not.

**Comment by jsurls** [comment]
@robbytx After looking at it further I see we were actually trying to create a distributable client in this project. So I'm all for lets not create a java one and replace it with a 'curl' equivalent if we can.

However I get the following errors when running the new 'hurl':
```
$ ./hurl -a apiKey -s secretKey http://localhost:8080/example
date: illegal option -- -
usage: date [-jnu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
```

Even entering the keys manually fail:
```
$ ./hurl http://localhost:8080/example
Enter Hurl ApiKey [****]: apiKey
Enter Hurl Secret [****]: secretKey
date: illegal option -- -
usage: date [-jnu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
```

**Comment by robbytx** [comment]
Ah... looks like I'm using ``date`` from GNU coreutils, instead of the native ``date`` that ships with Mac OS X.  I'll have to see how to get it working with the native version.

**Comment by robbytx** [comment]
@jsurls can you please fetch the updated script and try again?  I've confirmed that the new command line is compatible with /bin/date on my Mac, and with online references:
 * http://pubs.opengroup.org/onlinepubs/009604599/utilities/date.html
 * https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man3/strftime.3.html#//apple_ref/doc/man/3/strftime

**Comment by jsurls** [comment]
Looks great!

**Pull request merged**